### ISSUE TITLE
Doctrees cache causing build loop

### ISSUE THREAD

**Issue by cag** [executablebooks/sphinx-autobuild#67]
To reproduce, simply instrument a project with the ReadTheDocs [Getting Started](https://docs.readthedocs.io/en/latest/getting_started.html) guide, then try to use `sphinx-autobuild . _build/html`.

See the documentation about the default behavior for the doctree build output [here](http://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-d).

This causes projects to rebuild unnecessarily after the doctrees cache files are generated.

Also, if you force the cache to refresh the environment (for example, with the `-E` option), the autobuilder will build continuously.

A workaround for now is to also specify the `-d` option, setting it to `_build/doctrees` for example.

**Comment by steve-chavez** [comment]
Running into the same issue when trying to build these docs https://github.com/PostgREST/postgrest-docs,  it loops like 6 times making the refresh really slow,  also the `-d` work around didn't do it for me, I ran the command like:
```
sphinx-autobuild -d _build/doctrees -d _build/.doctrees . _build
```
@cag Any advice for making the `-d` option work?

**Comment by cag** [comment]
@steve-chavez Oddly enough, I ran `sphinx-autobuild . _build` on postgrest-docs and everything worked fine, with _no_ doctrees being generated at all. Running `sphinx-autobuild -d _build/doctrees . _build` generates doctrees, but also works fine. This is very strange... I'm not sure why this is happening.

**Comment by steve-chavez** [comment]
@cag There's a `build/.doctrees`(notice dot) generated though, also I've noticed that even when doing an `ls` the autobuild starts with a `.git/index.lock changed` message, and the `--ignore .git/` doesn't seem to be working.

**Comment by steve-chavez** [comment]
Some sensible defaults would be a good addition to the project. For now I've switched to plain livereload https://livereload.readthedocs.io/en/latest/, the sphinx example is working fine.

**Comment by aaronraimist** [comment]
> and the` --ignore .git/` doesn't seem to be working.

I know you found a solution but just adding a note here to say that the `--ignore` option is looking for either an exact match on the file name or a wildcard card character. So ` --ignore .git/` would only ignore changes to `.git/` itself. `--ignore .git/*` is what you want.

**Comment by pradyunsg** [comment]
Interesting! Looks like a reasonable bug to be fixed.

**Comment by pradyunsg** [comment]
This should be fixed by #88.

Could someone try the current master (`pip install https://github.com/GaretJax/sphinx-autobuild/archive/develop.zip`) and confirm whether this behavior is now fixed?

**Comment by pradyunsg** [comment]
Assuming that the silence means that this is fixed. If not, please file a new issue with clear reproduction instructions. :)
